clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -2016,7 +2016,25 @@ LValue CIRGenFunction::emitCastLValue(const CastExpr *E) {
   case CK_ToUnion:
     assert(0 && "NYI");
   case CK_BaseToDerived: {
-    assert(0 && "NYI");
+    const auto *DerivedClassTy = E->getType()->castAs<RecordType>();
```

Comment:
Please make sure to abide to https://llvm.github.io/clangir/GettingStarted/coding-guideline.html, `DerivedClassTy` -> `derivedClassTy`. Same for other variables, etc